### PR TITLE
389 Revamped Home Page

### LINKED ISSUE #389: Revamp Home Page
### Describe the Current Behavior/Feature:
To date, the Home Page has been minimal and lacking in information.

---
### Proposed Behavior/Feature:
Update the current Home Page (a box with that simply states "HOME PAGE") with a more detailed one.

---
### Rationale:
Home pages benefit from informing users, particularly first-time ones, about the background and features of the app. They also provide a welcoming and appealing initial experience.

---
### Proposed Implementation (if applicable):
The UX team is in the process of designing a home page. The assignee should consult with them before beginning work.

### PR DESCRIPTION
**Pull request by Raspber**
# Revamped Home Page

---
## This PR:
**1.** Added some basic images along with filler text so the home page is not empty.
**2.** Request demo button call to action.  
**3.** Moved assets to public folder

---
### Components

- Home.jsx
-HomeSection.jsx
-KeyFeatures.jsx

### Tests

No test files are affected

### Other Files

- app.png
- key-features-image.png
- web_security.png

---
## Screenshots (if applicable):
Current green theme with revamped home page
revamped-home-page

---
## Future Steps/PRs Needed to Finish This Work (optional):

---

### CODE REVIEW

File: src/pages/Home.jsx
Comment (leekahung) [inline src/pages/Home.jsx]: With MUI components, we're using the default breakpoints for responsive components, see example with the Typography component below this one with `xs`, `sm`, and `md`. Why not utilize those? Or if we need, why not just create custom breakpoints for MUI in `theme.js` to include these if we want to keep these breakpoints?

File: src/pages/Home.jsx
Comment (leekahung) [inline src/pages/Home.jsx]: Think we could replace this with margin: "18px 0"

File: src/pages/Home.jsx
Comment (leekahung) [inline src/pages/Home.jsx]: I don't think `maxWidth="s"` is a valid attribute for `Container`?

File: src/pages/Home.jsx
Comment (leekahung) [inline src/pages/Home.jsx]: I've noticed you're using `mb` for some `sx` attributes and `margin-bottom` for others. Think we should stick with one version to avoid confusion.

File: src/pages/Home.jsx
Comment (leekahung) [inline src/pages/Home.jsx:1]: Think we could split the contents within the Home page into three smaller components or one reusable component where you could place an image at the top, a title in the middle, and everything else below:

```
<Box>
  <img />
  <strong>title</strong>
  <other content>
</Box>
```

And use a `.map` to generate the three sections. Either approach would do.

File: src/theme.js
Comment (leekahung) [inline src/theme.js]: I think you don't need to state that it's temporary. If we wish to change the color we can simply change it to purple.

**Review by leekahung — CHANGES_REQUESTED** [review]
See feedback.

File: src/pages/Home.jsx
Comment (Raspber) [reply src/pages/Home.jsx]: Will add the break points to theme.js

File: src/pages/Home.jsx
Comment (Raspber) [reply src/pages/Home.jsx:1]: This sounds like a great idea. Ill do this.

**Comment by Raspber** [comment]
> See feedback.

Will make the changes required👍🏽

File: src/pages/Home.jsx
Comment (xscottxbrownx) [inline src/pages/Home.jsx]: should be `@memberof`

File: src/pages/Home.jsx
Comment (xscottxbrownx) [inline src/pages/Home.jsx]: is this not the same?

File: src/pages/Home.jsx
Comment (xscottxbrownx) [inline src/pages/Home.jsx]: not sure this is optimized... I'd have to look at our other code, but not sure if you need the `sm`.   Happens a lot in this file.

**Review by xscottxbrownx — CHANGES_REQUESTED** [review]
I back KaHung's comment on making the homepage reusable components. It'll be much easier to change styling/maintain in future.

---

Another issue is the images. We are going to run into the same issue we did with navbar logo.  

Maybe we need to merge KaHung's fix for that before this PR, then move these assets to that folder?

File: src/pages/Home.jsx
Comment (xscottxbrownx) [reply src/pages/Home.jsx]: What I mean by this is that I believe MUI to be "mobile-first" like Tailwind and Bootstrap. It may stay at that size until you tell it otherwise.  But again, I'm not positive.    I may be mixing it up with MUI media queries.

File: src/pages/Home.jsx
Comment (Raspber) [reply src/pages/Home.jsx]: I believe i have fixed this.

File: src/assets/app.png
Comment (leekahung) [inline src/assets/app.png:1]: The image assets should be moved to the new `public` directory so that they can be rendered in the production build. Previously, the image files did not render if we were to source from `src/assets` and the image links all broke in production even though it was fine on the development build.

This is more of a build thing by Vite, but it should be a quick fix. Check out the recent PR #407 bug fix for guidance on what to do.

File: src/components/Home/ReusableHomeSection.jsx
Comment (leekahung) [inline src/components/Home/ReusableHomeSection.jsx]: Create a new JSDoc for this new component. I think you could use the `DeleteClientModal.jsx` as a guide to how to write it up. (see https://github.com/codeforpdx/PASS/blob/Development/src/components/Modals/DeleteContactModal.jsx)

File: src/theme.js
Comment (leekahung) [inline src/theme.js]: Just `purple` should be fine

**Review by leekahung — DISMISSED** [review]
Think we're getting close to where we want. See feedback.

File: src/pages/Home.jsx
Comment (leekahung) [reply src/pages/Home.jsx]: I don't think this has been addressed yet?

File: src/pages/Home.jsx
Comment (leekahung) [inline src/pages/Home.jsx]: If `xs` and `sm` are the same, I think we only need to set one of them? Don't think we need 2.

I think the same could be said for the other media queries in this file.

File: src/pages/Home.jsx
Comment (leekahung) [reply src/pages/Home.jsx]: This has not been addressed.

File: src/components/Home/ReusableHomeSection.jsx
Comment (leekahung) [inline src/components/Home/ReusableHomeSection.jsx]: Think we could just adjust this to just `purple`?

File: src/pages/Home.jsx
Comment (leekahung) [inline src/pages/Home.jsx]: Think this entire block could be made into one line: `icon={<SecurityIcon sx={{ color: 'purple.main' }}/>}`

File: src/pages/Home.jsx
Comment (leekahung) [inline src/pages/Home.jsx]: This block could also be made into one line

File: src/pages/Home.jsx
Comment (leekahung) [inline src/pages/Home.jsx]: This block could also be made into one line

File: src/components/Home/ReusableHomeSection.jsx
Comment (leekahung) [inline src/components/Home/ReusableHomeSection.jsx]: Just noticed this now, but this should be a round bracket/parenthesis `(` since it's doing an implicit return for the new component.

File: src/components/Home/ReusableHomeSection.jsx
Comment (leekahung) [inline src/components/Home/ReusableHomeSection.jsx]: This should be the closing bracket/parenthesis `)` for the new component.

File: src/components/Home/ReusableHomeSection.jsx
Comment (leekahung) [inline src/components/Home/ReusableHomeSection.jsx]: the semi-colon is not needed

File: src/components/Home/ReusableHomeSection.jsx
Comment (leekahung) [reply src/components/Home/ReusableHomeSection.jsx]: This should fix the rendering problem that plagues this PR.

File: src/components/Home/ReusableKeyFeatures.jsx
Comment (leekahung) [inline src/components/Home/ReusableKeyFeatures.jsx]: This shares the same rendering issue as above, but could easily be fixed with the brackets/parentheses mentioned above for `ReusableHomeSection.jsx`

File: src/components/Home/ReusableHomeSection.jsx
Comment (leekahung) [inline src/components/Home/ReusableHomeSection.jsx:1]: Also, think we could simply call this `HomeSection.jsx`

File: src/components/Home/ReusableKeyFeatures.jsx
Comment (leekahung) [inline src/components/Home/ReusableKeyFeatures.jsx:1]: We could simply name this `KeyFeatures.jsx`

File: src/pages/Home.jsx
Comment (leekahung) [reply src/pages/Home.jsx]: Think this was what I mentioned with regards to using both `xs` and `sm`. Un-resolving this. I don't think this has been addressed?

File: src/components/Home/ReusableHomeSection.jsx
Comment (xscottxbrownx) [reply src/components/Home/ReusableHomeSection.jsx:1]: Agreed. All React components are inherently reusable.

File: src/components/Home/ReusableHomeSection.jsx
Comment (xscottxbrownx) [inline src/components/Home/ReusableHomeSection.jsx]: Import these individually (like our other components do) to keep the package size down.

File: src/components/Home/ReusableHomeSection.jsx
Comment (xscottxbrownx) [inline src/components/Home/ReusableHomeSection.jsx]: don't think the `md` is necessary. This should render at 1.5rem from `sm` up.

---

Side note:  maybe it's necessary here due to the amount of text, but we haven't had to do font sizes like this yet elsewhere.   The `Typography` component has different variations you can set (like h1-h6, body, etc.)  See [here](https://mui.com/material-ui/react-typography/#component) and [here](https://mui.com/material-ui/react-typography/#changing-the-semantic-element) for more info - if you haven't already.

File: src/components/Home/ReusableKeyFeatures.jsx
Comment (xscottxbrownx) [inline src/components/Home/ReusableKeyFeatures.jsx]: for MUI use default imports vs named imports... their documentation talked about performance issues/package size/etc... 

so these should be more like: `import Box from '@mui/material/Box';` 

see `Home.jsx` for more examples.  It's a little annoying as a developer, instead of just being able to use the named import unfortunately.

File: src/pages/Home.jsx
Comment (xscottxbrownx) [inline src/pages/Home.jsx]: since all these icons that are passed to `ReusableKeyFeatures` are `color: 'Purple.main'`
**&**
in `ReusableKeyFeatures` the `Typography`, located in the same `Box` as this icon, is also the same color

Then, we should just set the `Box` in `ReusableKeyFeatures` with `color: 'Purple.main'` and so we don't have to do this each time here (or on the `Typography` within the box either.)

File: src/components/Home/ReusableKeyFeatures.jsx
Comment (xscottxbrownx) [inline src/components/Home/ReusableKeyFeatures.jsx]: See my comment in the `Home.jsx` file of setting color for `Box` here, so we don't have to set it for each icon and this `Typography`.

**Review by xscottxbrownx — CHANGES_REQUESTED** [review]

File: src/components/Home/ReusableKeyFeatures.jsx
Comment (xscottxbrownx) [reply src/components/Home/ReusableKeyFeatures.jsx:1]: agreed, same reason as above.

File: src/pages/Home.jsx
Comment (xscottxbrownx) [reply src/pages/Home.jsx]: See my comment on `Home.jsx` to make this even shorter here.

**Comment by andycwilliams** [comment]
I personally like making purple the main color. However, in my meetings with the UX team, it's become clear that choosing a theme/color palette is actually much more complicated than we might have assumed. As part of our overall plan, finding a new theme is a lower priority. So in my opinion we can hold off on changing that for the time being.

File: src/components/Home/ReusableHomeSection.jsx
Comment (andycwilliams) [reply src/components/Home/ReusableHomeSection.jsx]: @Raspber Are you running the prettier script before pushing commits? If not, that should clear up a lot of these minor issues. If available, you could also add prettier as an extension in your code editor. That's what I do.

**Comment by Raspber** [comment]
> I personally like making purple the main color. However, in my meetings with the UX team, it's become clear that choosing a theme/color palette is actually much more complicated than we might have assumed. As part of our overall plan, finding a new theme is a lower priority. So in my opinion we can hold off on changing that for the time being.



> I personally like making purple the main color. However, in my meetings with the UX team, it's become clear that choosing a theme/color palette is actually much more complicated than we might have assumed. As part of our overall plan, finding a new theme is a lower priority. So in my opinion we can hold off on changing that for the time being.

Deleted the purple per request.

File: src/components/Home/ReusableKeyFeatures.jsx
Comment (Raspber) [reply src/components/Home/ReusableKeyFeatures.jsx]: sent a private pm in regards to this. @xscottxbrownx @leekahung

**Comment by xscottxbrownx** [comment]
@Raspber - will you still have access to these graphics after merging into codebase?  

Debating if we should add the purple images in the codebase or not, since we aren't using yet - and not sure if that's what team will land on.

**Comment by Raspber** [comment]
> @Raspber - will you still have access to these graphics after merging into codebase?  
> 
> Debating if we should add the purple images in the codebase or not, since we aren't using yet - and not sure if that's what team will land on. 

Yes, i do have access to these images. I think I can get rid of the purple images because ux team is most likely going to come up with a brand new home page. Ka Hung just wanted something in there so it wasnt so plain. In less the ux team may want to use some of these images.

**Comment by andycwilliams** [comment]
> @Raspber - will you still have access to these graphics after merging into codebase?
> 
> Debating if we should add the purple images in the codebase or not, since we aren't using yet - and not sure if that's what team will land on.

We can talk more about it at the meeting but I think we shouldn't include anything until we know for sure that we need it. And yeah it all depends on the UX team anyhow so I'd wait at least until we talk to them.

**Comment by xscottxbrownx** [comment]
@Raspber at the meetup we agreed, if you can delete the purple image files from this - we can approve and merge.

**Comment by xscottxbrownx** [comment]
We don't need the images in both `public` and `src` folders... 

I did ask for help at last night's meeting to research what the deal is for vite and proper way to use images. So, hopefully we can get a "correct" answer and procedure for us.

**Comment by ogorman89** [comment]
> I did ask for help at last night's meeting to research what the deal is for vite and proper way to use images. So, hopefully we can get a "correct" answer and procedure for us.

@xscottxbrownx The images are loading fine from the `public` folder when I build locally. Was there something else I needed to do to reproduce the issue? 

My understanding of your comment is that we just want to figure out the best practice for storing these assets going forward. 

Edit: referenced incorrect folder.

**Comment by xscottxbrownx** [comment]
> > I did ask for help at last night's meeting to research what the deal is for vite and proper way to use images. So, hopefully we can get a "correct" answer and procedure for us.
> 
> @xscottxbrownx The images are loading fine from the `public` folder when I build locally. Was there something else I needed to do to reproduce the issue?
> 
> My understanding of your comment is that we just want to figure out the best practice for storing these assets going forward.
> 
> Edit: referenced incorrect folder.

Yes photos and actually the `HomeSection` components are loading fine now because we fixed the actual issue which was improper destructuring of props being passed to the component.

But now 2 things...

For this PR
1) photos have been put in both `src` and `public` folders, just needs to be one.  Right now they are being loaded from `src/assets` as you can see in `Home.jsx` - but there was a past PR that Ka Hung ended up adding the `public` folder instead because although it loaded on Development, there were issues on Master.

In General
2) need to research what is proper way to use/store/access images and paths in Vite.  So, we can move forward in future with no issues.   My 5 second research was that `public` folder was one way they could be handled.   Just need to figure it out, and then be consistent from there forward.

**Comment by leekahung** [comment]
Yeah, it seems like Vite doesn't pick up the source files for images under src/assets in the production build (see issue #406). Images end up breaking there even though they show up locally in the development build.

Having the image files stored in the public directory is one way to have static assets built with the site. (See link in issue #406 for Vite documentation and PR #407 on how I implement it here on PASS)

Since that's the case, I agree with Scott. It'll be good to  have them all inside public. Though I wonder if we make folders inside public mainly for images like public/assets or public/images and place it in them instead of the root of public (have not tried it yet in the project).

**Comment by andycwilliams** [comment]
For what it's worth, the professional projects I've seen have had images in the `public` folder, so that seems to be best practice anyhow.

Also, having folders inside of `public` ought to work fine. Though I haven't experimented very much with it.

Otherwise, we're very close to merging this, no?

**Comment by xscottxbrownx** [comment]
> Otherwise, we're very close to merging this, no?

Yes, we just need to:
 1. get rid of the assets from the `src` folder
 2. put in the public path for the image src's in the `Home` component instead of the source path.

**Review by xscottxbrownx — CHANGES_REQUESTED** [review]
See my last comment responding to Andy... two things to address.

**Comment by xscottxbrownx** [comment]
@Raspber why doing this here on line 43, but not on line 53 and 60?

Screenshot 2023-09-23 at 6 04 42 AM

The vite documentation [here](https://vitejs.dev/guide/assets.html#the-public-directory) says it should be `/assets/filename.fileextenstion` (eventhough it seems to work either way.)

--------

@leekahung do you know of anybody still currently working on the driver's license barcode data capture idea?  This PR currently deletes the `barcode demo` from `src/assets` (probably because I wasn't clear enough in my number 2 above in my last comment), and not sure if we should delete that yet.

**Comment by leekahung** [comment]
> @leekahung do you know of anybody still currently working on the driver's license barcode data capture idea?  This PR currently deletes the `barcode demo` from `src/assets` (probably because I wasn't clear enough in my number 2 above in my last comment), and not sure if we should delete that yet.

At the moment, no. Think Brian and Jared were the people working on them before. With regards to the barcode demo file, it's simply there to help developers test out the feature initially since it's saved with the repo. But it's not really necessary, I think.

**Comment by Raspber** [comment]
> @Raspber why doing this here on line 43, but not on line 53 and 60?
> 
> Screenshot 2023-09-23 at 6 04 42 AM
> 
> The vite documentation [here](https://vitejs.dev/guide/assets.html#the-public-directory) says it should be `/assets/filename.fileextenstion` (eventhough it seems to work either way.)
> 
> --------
> 
> @leekahung do you know of anybody still currently working on the driver's license barcode data capture idea?  This PR currently deletes the `barcode demo` from `src/assets` (probably because I wasn't clear enough in my number 2 above in my last comment), and not sure if we should delete that yet.
> 

Did not see that. Thank you. Will get it removed. I can also add the barcode image if youd like.

**Comment by xscottxbrownx** [comment]
@Raspber - all good.  Fixed. Merging now.

Great work on all this!

**Review by xscottxbrownx — APPROVED** [review]
All issues addressed

**Comment by Raspber** [comment]
> @Raspber - all good.  Fixed. Merging now.
> 
> Great work on all this!

Thanks for looking out! Its appreciated!

**Comment by leekahung** [comment]
Could you update the PR details with the changes included @Raspber? Thanks!

**Pull request merged**
